common/src/main/java/io/netty/util/ResourceLeakDetectorFactory.java

```diff
@@ -61,60 +61,78 @@ public static void setResourceLeakDetectorFactory(ResourceLeakDetectorFactory fa
      * @param <T> - the type of the resource class
      * @return - a new instance of {@link ResourceLeakDetector}
      */
-    public abstract <T> ResourceLeakDetector<T> newResourceLeakDetector(final Class<T> resource);
+    public final <T> ResourceLeakDetector<T> newResourceLeakDetector(Class<T> resource) {
+        return newResourceLeakDetector(resource, ResourceLeakDetector.DEFAULT_SAMPLING_INTERVAL, Long.MAX_VALUE);
+    }
+
+    /**
+     * Returns a new instance of a {@link ResourceLeakDetector} with the given resource class.
+     *
+     * @param resource - the resource class used to initialize the {@link ResourceLeakDetector}
+     * @param samplingInterval - the interval on which sampling takes place
+     * @param maxActive - the maximum active instances
+     * @param <T> - the type of the resource class
+     * @return - a new instance of {@link ResourceLeakDetector}
+     */
+    public abstract <T> ResourceLeakDetector<T> newResourceLeakDetector(
+            Class<T> resource, int samplingInterval, long maxActive);
 
     /**
      * Default implementation that loads custom leak detector via system property
      */
     private static final class DefaultResourceLeakDetectorFactory extends ResourceLeakDetectorFactory {
-
-        private final String customLeakDetector;
-        private final Constructor customClassConstructor;
+        private final Constructor<?> customClassConstructor;
 
         public DefaultResourceLeakDetectorFactory() {
-            this.customLeakDetector = AccessController.doPrivileged(new PrivilegedAction<String>() {
-                @Override
-                public String run() {
-                    return SystemPropertyUtil.get("io.netty.customResourceLeakDetector");
-                }
-            });
-
-            this.customClassConstructor = customClassConstructor();
+            customClassConstructor = customClassConstructor(
+                    AccessController.doPrivileged(new PrivilegedAction<String>() {
```

Comment:
This method will propagate runtime exceptions, and `SystemPropertyUtil.get` can throw a `SecurityException` ... should we surround this in a try/catch?